src/components/switch/switch.vue

```diff
@@ -56,11 +60,17 @@
                     return false;
                 }
 
-                const checked = !this.currentValue;
-                this.currentValue = checked;
-                this.$emit('input', checked);
-                this.$emit('on-change', checked);
-                this.dispatch('FormItem', 'on-form-change', checked);
+                checkPromise(this.beforeChange(this.currentValue)).then((result) => {
```

Comment:
Not sure I follow the logic here. If `beforeChange` is `null` then we get a error right?
Maybe should be 

    checkPromise(this.beforeChange)(this.currentValue).then((result) => {

?

Reply:
My bad. I didn't fully understand how promise works at that point. I'll fix it soon.

Reply:
@rijn cool no problem. Maybe we will do this in more components later.